### PR TITLE
Fix when NPM view is shown

### LINKED ISSUE #104591: There is no way to show the NPM scripts explorer when there is no top level package.json
Issue Type: <b>Bug</b>

Previous to this version, NPM Script Explorer would show for the following scenario:

Solution
    - Project1
        - package.json (with scripts)
    - Project2
        - package.json (with scripts)
    - Project3
        - no package.json

To test if this was an extension causing this issue, I installed the latest insider:
(Version: 1.48.0-insider (user setup)
Commit: db40434f562994116e5b21c24015a2e40b2504e6
Date: 2020-08-11T23:21:50.282Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041)

I made sure that no extensions were installed and that NPM Script Explorer was turned on. When I opened up the solution, the NPM Script Explorer did not show up.

If I open up by project, the NPM Script Explorer appears as expected.

I checked the Dev Tools and there were no errors in the console.

VS Code version: Code 1.48.0 (db40434f562994116e5b21c24015a2e40b2504e6, 2020-08-13T07:50:42.600Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2808)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.83GB (16.61GB free)|
|Process Argv|--folder-uri file:///c%3A/dev/Repos/site-dental-ecom|
|Screen Reader|no|
|VM|0%|
</details>

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #104591

### CODE REVIEW

**Comment by IllusionMH** [comment]
Does it make sense to have it all the time even for Python, Java users?

There are checks like this that won't show this view (in Explorer views list) only if there are files that can contain tasks
https://github.com/microsoft/vscode/blob/bb3c233e97d8e2d78d9fa97a5fea5274e8300ad1/extensions/npm/src/tasks.ts#L148-L149

Maybe it make sense to use it as condition? Or this checks is too costly for extensions startup?

**Comment by alexr00** [comment]
@IllusionMH I think it's ok to have always be in the Views menu. Having it in the Views menu is sufficiently out of the way that most people won't feel like it is unnecessarily in their face.

**Comment by IllusionMH** [comment]
@alexr00 OK, I see your point.

**Review by crypt096 — APPROVED** [review]

File: extensions/npm/package.json [hunk 1]
```diff
@@ -237,6 +236,7 @@
           "type": "boolean",
           "default": false,
           "scope": "resource",
+          "deprecationMessage": "The NPM Script Explorer is now available in 'Views' menu in the Explorer in all folders.",
```
Comment (egamma) [inline extensions/npm/package.json:239]: Wording? The NPM Script Explorer is now available in **the** 'Views' menu in the Explorer in all folders."

File: extensions/npm/src/npmMain.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import * as vscode from 'vscode';
 import { addJSONProviders } from './features/jsonContributions';
 import { runSelectedScript, selectAndRunScriptFromFolder } from './commands';
 import { NpmScriptsTreeDataProvider } from './npmView';
-import { invalidateTasksCache, NpmTaskProvider, hasPackageJson } from './tasks';
```
Comment (egamma) [inline extensions/npm/src/npmMain.ts:11]: I think this is the only use of the `hasPackageJson` utility function and it could be deleted now.

**Review by egamma — APPROVED** [review]

**Comment by egamma** [comment]
LGTM some small tweaks as comments above.

File: extensions/npm/package.json (hunk 1 above)
Comment (alexr00) [reply extensions/npm/package.json:239]: Fixed, thank you for catching.

File: extensions/npm/src/npmMain.ts (hunk 2 above)
Comment (alexr00) [reply extensions/npm/src/npmMain.ts:11]: 👍

**Pull request merged**